corehq/apps/hqadmin/reports.py

```diff
@@ -333,3 +338,157 @@ def _shorten_and_hyperlink_commit(self, commit_sha):
                 abbrev_sha=commit_sha[:7]
             )
         return None
+
+
+class UCRRebuildRestrictionTable:
+    UCR_RESTRICTION_THRESHOLD = 1_000_000
+
+    restriction_ff_status: str
+
+    def __init__(self, *args, **kwargs):
+        self.restriction_ff_status = kwargs.get('restriction_ff_status')
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(gettext_lazy("Domain")),
+            DataTablesColumn(gettext_lazy("Case count")),
+            DataTablesColumn(gettext_lazy("Form count")),
+            DataTablesColumn(gettext_lazy("UCR rebuild restriction status")),
+        )
+
+    @property
+    def rows(self):
+        rows = []
+
+        ucr_domains = self.ucr_domains
+        if not ucr_domains:
+            return []
+
+        case_count_by_domain = self._case_count_by_domain(ucr_domains)
+        form_count_by_domain = self._forms_count_by_domain(ucr_domains)
+
+        for domain in ucr_domains:
+            case_count = getattr(case_count_by_domain.get(domain), 'doc_count', 0)
+            form_count = getattr(form_count_by_domain.get(domain), 'doc_count', 0)
+
+            if self.should_show_domain(domain, case_count, form_count):
+                rows.append(
+                    self._row_data(domain, case_count, form_count)
+                )
+
+        return rows
+
+    @property
+    def ucr_domains(self):
```

Comment:
Would it be possible for us to make this `memoized`? Then we only need to retrieve the list of UCR domains once.

Reply:
Good idea